### ISSUE TITLE
Marking the offset per partition

### ISSUE THREAD

**Issue by tarswars** [bsm/sarama-cluster#28]
Dear dim,

Your library works really well but I needed a way to mark the offset per partition (as different partitions could run at a different paste and I want ONLY message constructed right before midnight to be committed per partition).

I've added the following function into the lib, this might be handy for your lib as well?
Could you also have a look if this won't cause any race conditions or other unexpected behavior?

```
func (c *Consumer) MarkOffsetPerPartition(topic string, part int32, offset int64) {
	c.subs.Fetch(topic, part).MarkOffset(offset, "")
}
```

Thanks in advance,
tarswars

**Comment by dim** [comment]
Take a look at the [existing method](https://github.com/bsm/sarama-cluster/blob/master/consumer.go#L101), you could just call `MarkOffset(&sarama.ConsumerMessage{Topic: topic, Partition: part, Offset: offset})` to achieve the same behaviour, no?

**Comment by tarswars** [comment]
Yes, that is correct. I should have elaborated a bit more.
In my case I wanted to give the previous offset

**Comment by dim** [comment]
Fixed via 44878dd3240f0fdd8dcc01e251b50b064ea68fd6
